syntax = "proto3";
option java_multiple_files = true;
option java_outer_classname = "MissionProto";
import public "Tracking.proto";
package avtm.center.proto;

service MissionService {
  rpc create(MissionRequest) returns (MissionResponse);
  rpc update(MissionUpdateStatusRequest) returns (Empty);
  rpc updateWayPoints(MissionUpdateStatusAndCoordinatesRequest) returns (Empty);
}

message MissionRequest {
  string deviceUuid = 1;
  string flightRequestUuid = 2;
  repeated Point coordinates = 3;
}

message MissionResponse {
  string missionUuid = 1;
  MissionStatus status = 2;
  bool is_valid = 3;
  repeated Point coordinates = 4;
  string deviceUuid = 5;
  string flightRequestUuid = 6;
}

message MissionUpdateStatusRequest {
  string missionUuid = 1;
  MissionStatus status = 2;
}

message MissionUpdateStatusAndCoordinatesRequest {
  string missionUuid = 1;
  MissionStatus status = 2;
  repeated Point coordinates = 3;
}

enum MissionStatus {
  CREATED = 0;
  IN_MOVE = 1;
  LANDED = 2;
}
